arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,360 @@
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: UITableViewCell
+        let phase: AGSCodedValue
+        switch Section.allCases[indexPath.section] {
+        case .included:
+            cell = tableView.dequeueReusableCell(withIdentifier: Section.included.cellIdentifier, for: indexPath)
+            phase = includedPhases[indexPath.row]
+            cell.textLabel?.text = "Phase: " + phase.name
+            if let summary = summaries[phase] {
+                cell.detailTextLabel?.text = "C: " + numberFormatter.string(from: NSNumber(value: summary.totalCustomers))! + "    L: " + numberFormatter.string(from: NSNumber(value: summary.totalLoad))!
+            } else {
+                cell.detailTextLabel?.text = "N/A"
+            }
+        case .excluded:
+            cell = tableView.dequeueReusableCell(withIdentifier: Section.excluded.cellIdentifier, for: indexPath)
+            phase = excludedPhases[indexPath.row]
+            cell.textLabel?.text = phase.name
+        }
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, editingStyleForRowAt indexPath: IndexPath) -> UITableViewCell.EditingStyle {
+        switch Section.allCases[indexPath.section] {
+        case .included:
+            return .delete
+        case .excluded:
+            return .insert
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        tableView.beginUpdates()
+        switch editingStyle {
+        case .delete:
+            tableView.deleteRows(at: [indexPath], with: .automatic)
+            let phase = includedPhases[indexPath.row]
+            includedPhases.remove(at: indexPath.row)
+            // Binary search to find the insertion index.
+            let insertionIndex = (excludedPhases as NSArray).index(
+                of: phase,
+                inSortedRange: NSRange(excludedPhases.indices),
+                options: [.insertionIndex],
+                usingComparator: { ($0 as! AGSCodedValue).name.compare(($1 as! AGSCodedValue).name) }
+            )
+            excludedPhases.insert(phase, at: insertionIndex)
+            let section = Section.excluded.rawValue
+            tableView.insertRows(at: [IndexPath(row: insertionIndex, section: section)], with: .automatic)
+        case .insert:
+            tableView.deleteRows(at: [indexPath], with: .automatic)
+            let phase = excludedPhases[indexPath.row]
+            excludedPhases.remove(at: indexPath.row)
+            includedPhases.append(phase)
+            let section = Section.included.rawValue
+            let newRow = tableView.numberOfRows(inSection: section)
+            tableView.insertRows(at: [IndexPath(row: newRow, section: section)], with: .automatic)
+        default:
+            break
+        }
+        tableView.endUpdates()
```

Comment:
I recommend putting this in a `defer`, right after the call to `beginUpdates()`:

```swift
tableView.beginUpdates()
defer { tableView.endUpdates() }
```

That opens up options for returning in the `default` case and then moving some of the logic after the `switch` statement. For example:

```swift
let phase = includedPhases[indexPath.row]
let indexPathToInsert: IndexPath
switch editingStyle { ... }
tableView.deleteRows(at: [indexPath], with: .automatic)
tableView.insertRows(at: [indexPathToInsert], with: .automatic)
```